src/steps/format-json.js

```diff
@@ -0,0 +1,53 @@
+import dasherize from "dasherize";
+import camelize from "../util/camelize";
+import Resource from "../types/Resource";
+import Collection from "../types/Collection";
+
+export function dasherizeResource(resource, registry) {
+  let exceptions = registry.behaviors(resource.type).dasherizeOutput.exceptions;
+  return transformResourceKeys(resource, dasherize, exceptions);
+}
+
+export function dasherizeResourceOrCollection(toDasherize, registry) {
+  return transformResourceOrCollection(toDasherize, dasherizeResource, registry);
+}
+
+export function camelizeResource(resource, registry) {
+  let exceptions = registry.behaviors(resource.type).dasherizeOutput._inverseExceptions;
+  return transformResourceKeys(resource, camelize, exceptions);
+}
+
+export function camelizeResourceOrCollection(toCamelize, registry) {
+  return transformResourceOrCollection(toCamelize, camelizeResource, registry);
+}
+
+function transformResourceKeys(resource, transformFn, exceptions) {
```

Comment:
A relationship object's `data` only contains `type` and `id`, because `data` contains "relationship identifier objects", which are just the `type`, `id` pairs. But the _name_ of each relationship, i.e. the keys of the "relationship_s_ object" do need to be camelized. Those relationship names are the keys of `resource.relationships`